### PR TITLE
fix(scale): Use *int for scale index

### PR DESCRIPTION
**Pull request by kanishkatn**
## Changes

Changed the type of scaleIndex from `*string` to `*int`.

We have two options to solve the issue of string index comparison.
- Convert the string index to int at the time of comparison
- Store the int index instead of the string index

I decided to go with the latter since it will allow us to validate the scaleIndex before encoding/decoding a struct.

## Tests

```sh
go test pkg/scale
```

## Issues

#3268 

## Primary Reviewer

@timwu20

### CODE REVIEW

**Comment by kanishkatn** [comment]
The test `TestStableNetworkRPC`, which isn't even closely related to this PR is failing for some reason, but it works in my machine. I'm debugging it.

**Review by EclesioMeloJunior — APPROVED** [review]

**Review by dimartiro — APPROVED** [review]

**Review by kishansagathiya — APPROVED** [review]

**Comment by kishansagathiya** [comment]
Could put `Fixes #issuenumber` to automatically close the issue on merging this issue.

File: pkg/scale/encode_test.go [hunk 1]
```diff
@@ -1196,8 +1213,12 @@ func Test_marshal_optionality_nil_cases(t *testing.T) {
...
 			name: t.name,
 			// in:      t.in,
-			wantErr: t.wantErr,
-			want:    t.want,
+
+			// these do not matter since we are testing nil cases
+			// and the want is being set to empty byte slice below
+			// marshal will not error out for nil cases since it never hits a type's marshal method
+			//wantErr: t.wantErr,
+			//want: t.want,
```
[the remark this answers is not in the document]
Comment (kanishkatn) [reply pkg/scale/encode_test.go:1221]: The comment was there to help you with the review. I'll be removing these.

**Comment by kanishkatn** [comment]
> LGTM! I can't recall why I didn't do this. Maybe I assumed I'd support something other than just the integer index as part of the scale struct tag.

I see. We could still extend it if we want to add anything else along with the scaleIndex in the tag!

**Pull request merged**